titus-server-master/src/main/java/com/netflix/titus/master/scheduler/resourcecache/OpportunisticCpuAllocation.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.master.scheduler.resourcecache;
+
+import java.util.Objects;
+
+public class OpportunisticCpuAllocation {
+    private final String taskId;
+    private final String agentId;
+    private final String allocationId;
+    private final int cpuCount;
+
+    public OpportunisticCpuAllocation(String taskId, String agentId, String allocationId, int cpuCount) {
```

Comment:
private as we have the builder?

Reply:
I considered, but the one-shot constructor seemed a good fit in `TaskCacheEventListener` where we didn't need to build the POJO incrementally